xls/passes/select_simplification_pass.cc

```diff
@@ -964,6 +966,41 @@ absl::StatusOr<bool> SimplifyNode(Node* node, const QueryEngine& query_engine,
     }
   }
 
+  // Specialize the arms of the select based on the selector value which chooses
+  // the arm.
+  if (node->Is<Select>()) {
+    Select* sel = node->As<Select>();
+    auto selr_width = sel->selector()->BitCountOrDie();
+    bool changed = false;
+    for (uint64 selr_value = 0; selr_value < sel->cases().size();
+         ++selr_value) {
+      Node* arm = sel->cases()[selr_value];
+      // Skip any cases where the select arm is used other than just in the
+      // select statement
+      if (arm->users().size() != 1) {
+        continue;
+      }
+      for (Node* selr_use : sel->selector()->users()) {
+        if (!post_dominator_analysis.NodeIsPostDominatedBy(selr_use, arm)) {
+          continue;
+        }
+        for (int64 operand_no = 0; operand_no < selr_use->operand_count();
```

Comment:
Rather than this loop, you can create the literal then call:

selr_use->ReplaceOperand(selr, literal);

Node::ReplaceOperand could use a better comment. Mind adding something like "Replace all instances of old_operand with new_operand." in that method comment? Thanks!